plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c

```diff
@@ -593,8 +593,12 @@ static obs_properties_t *vaapi_properties(void *unused)
 			    strcmp(file_name, "..") == 0)
 				continue;
 
-			char path[64] = "\0";
-			sprintf(path, "/dev/dri/by-path/%s", file_name);
+			char path[64];
+			char path_suffix[47];
+
+			strncpy(path_suffix, file_name, 47);
```

Comment:
I think useless copies make these functions harder to understand. Google suggests branching on thbe result of snprintf will calm gcc, if that works then lets blog that you have a malformed entry and continue to skip this.

Reply:
I switch to `snprintf` and "used" the result to prevent the overflow warning, since the `file_name` prevent to go beyond a length of 64.